### PR TITLE
Merge develop into 2.0

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: bindings/nodejs/lib/utils/utils.ts [hunk 1]
```diff
@@ -122,24 +136,25 @@ export class Utils {
...
-     * Computes the required storage deposit of an output.
+     * Compute the required storage deposit of an output.
+     *
      * @param output The output.
      * @param rent Rent cost of objects which take node resources.
      * @returns The required storage deposit.
      */
-    static computeStorageDeposit(output: Output, rent: IRent): u64 {
-        const depositHex = callUtilsMethod({
+    static computeStorageDeposit(output: Output, rent: IRent): bigint {
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/utils/utils.ts]: Is this supposed to stay u64?

File: bindings/nodejs/lib/wallet/wallet.ts [hunk 2]
```diff
@@ -145,6 +148,17 @@ export class Wallet {
         return this.methodHandler.getClient();
     }
 
+    /**
+     * Get chrysalis data.
+     */
+    async getChrysalisData(): Promise<Record<string, string>> {
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/wallet/wallet.ts]: Delete

File: bindings/python/README.md [hunk 3]
```diff
@@ -86,21 +86,21 @@ that connects to the [Shimmer Testnet](https://api.testnet.shimmer.network) usin
...
+    "wallet.stronghold", "some_hopefully_secure_password")
 
-wallet = Wallet('./alice-walletdb', wallet_options, coin_type=CoinType.SHIMMER, secret_manager)
+wallet = Wallet('./alice-walletdb', client_options,
+                CoinType.SHIMMER, secret_manager)
 
 # Store the mnemonic in the Stronghold snapshot. This only needs to be done once
 account = wallet.store_mnemonic("flame fever pig forward exact dash body idea link scrub tennis minute " +
-          "surge unaware prosper over waste kitten ceiling human knife arch situate civil")
+                                "surge unaware prosper over waste kitten ceiling human knife arch situate civil")
```
Comment (DaughterOfMars) [inline bindings/python/README.md:103]: not sure bout this

File: bindings/wasm/lib/bindings.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { __UtilsMethods__ } from './utils';
 // Import needs to be in a single line, otherwise it breaks
 // prettier-ignore
 // @ts-ignore: path is set to match runtime transpiled js path when bundled.
-import { initLogger, createClient, destroyClient, createSecretManager, createWallet, callClientMethodAsync, callSecretManagerMethodAsync, callUtilsMethodRust, callWalletMethodAsync, destroyWallet, listenWalletAsync, getClientFromWallet, getSecretManagerFromWallet, listenMqtt, migrateStrongholdSnapshotV2ToV3 } from '../wasm/iota_sdk_wasm';
+import { initLogger, createClient, destroyClient, createSecretManager, createWallet, callClientMethodAsync, callSecretManagerMethodAsync, callUtilsMethodRust, callWalletMethodAsync, destroyWallet, listenWalletAsync, getClientFromWallet, getSecretManagerFromWallet, listenMqtt, migrateStrongholdSnapshotV2ToV3, migrateDbChrysalisToStardust } from '../wasm/iota_sdk_wasm';
```
Comment (DaughterOfMars) [inline bindings/wasm/lib/bindings.ts]: delete

File: bindings/nodejs/lib/wallet/wallet.ts (hunk 2 above)
Comment (thibault-martinez) [reply bindings/nodejs/lib/wallet/wallet.ts]: Removed locally, no need to further comment about it

File: bindings/wasm/src/wallet.rs [hunk 5]
```diff
@@ -141,3 +141,14 @@ pub async fn listen_wallet(
 
     Ok(JsValue::UNDEFINED)
 }
+
+/// Rocksdb chrysalis migration is not supported for WebAssembly bindings.
```
Comment (DaughterOfMars) [inline bindings/wasm/src/wallet.rs]: delete

File: bindings/wasm/lib/bindings.ts (hunk 4 above)
Comment (thibault-martinez) [reply bindings/wasm/lib/bindings.ts]: Removed locally, no need to further comment about it

File: bindings/wasm/src/wallet.rs (hunk 5 above)
Comment (thibault-martinez) [reply bindings/wasm/src/wallet.rs]: Removed locally, no need to further comment about it

File: cli/README.md [hunk 6]
```diff
@@ -37,7 +35,7 @@ Alternatively, you can select an existing account by its alias:
 
 ## Commands
 
-To see the full list of available commands look at the documentation [here](./documentation/docs).
+To see the full list of available commands look at the documentation [here](https://wiki.iota.org/shimmer/cli-wallet/welcome/).
```
Comment (DaughterOfMars) [inline cli/README.md:38]: These changes right?

File: bindings/python/README.md (hunk 3 above)
Comment (thibault-martinez) [reply bindings/python/README.md:103]: Idk, I guess it was done in develop 🤷🏻‍♂️

File: cli/README.md (hunk 6 above)
Comment (thibault-martinez) [reply cli/README.md:38]: Yes

File: sdk/src/client/secret/mod.rs [hunk 7]
```diff
@@ -3,12 +3,17 @@
...
 #[cfg_attr(docsrs, doc(cfg(feature = "ledger_nano")))]
 pub mod ledger_nano;
-/// Module for signing with a mnemonic or seed
+/// Module for mnemonic based secret management.
 pub mod mnemonic;
-/// Module for signing with a Stronghold vault
+/// Module for single private key based secret management.
+#[cfg(feature = "private_key_secret_manager")]
+#[cfg_attr(docsrs, doc(cfg(feature = "private_key_secret_manager")))]
+pub mod private_key;
```
Comment (DaughterOfMars) [inline sdk/src/client/secret/mod.rs:15]: Create an issue to add the error(s) for this?

Comment (thibault-martinez) [reply sdk/src/client/secret/mod.rs:15]: https://github.com/iotaledger/iota-sdk/issues/1105

File: sdk/src/types/api/core/response.rs
Comment (DaughterOfMars) [inline sdk/src/types/api/core/response.rs:1]: There are still a lot of remaining `cfg_attr` tags that can be removed

File: sdk/src/wallet/core/operations/storage.rs [hunk 8]
```diff
@@ -84,6 +84,14 @@ mod storage_stub {
             Ok(res.map(Into::into))
         }
     }
+
+    impl Wallet {
+        pub async fn get_chrysalis_data(
```
Comment (DaughterOfMars) [inline sdk/src/wallet/core/operations/storage.rs]: delete

File: sdk/src/wallet/storage/constants.rs [hunk 9]
```diff
@@ -31,3 +31,5 @@ pub(crate) const DATABASE_SCHEMA_VERSION_KEY: &str = "database-schema-version";
 pub(crate) const PARTICIPATION_EVENTS: &str = "participation-events";
 #[cfg(feature = "participation")]
 pub(crate) const PARTICIPATION_CACHED_OUTPUTS: &str = "participation-cached-outputs";
+
+pub(crate) const CHRYSALIS_STORAGE_KEY: &str = "chrysalis-data";
```
Comment (DaughterOfMars) [inline sdk/src/wallet/storage/constants.rs]: delete

File: sdk/src/wallet/core/operations/storage.rs (hunk 8 above)
Comment (thibault-martinez) [reply sdk/src/wallet/core/operations/storage.rs]: Removed locally, no need to further comment about it

File: sdk/src/wallet/storage/constants.rs (hunk 9 above)
Comment (thibault-martinez) [reply sdk/src/wallet/storage/constants.rs]: Removed locally, no need to further comment about it

File: bindings/nodejs/lib/utils/utils.ts (hunk 1 above)
Comment (thibault-martinez) [reply bindings/nodejs/lib/utils/utils.ts]: Reverted

File: bindings/core/src/response.rs [hunk 10]
```diff
@@ -1,6 +1,7 @@
 // Copyright 2023 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+use std::collections::HashMap;
```
Comment (DaughterOfMars) [inline bindings/core/src/response.rs]: Just revert?

Comment (thibault-martinez) [reply bindings/core/src/response.rs]: Done

**Review by DaughterOfMars — APPROVED** [review]
I think!! it's okay??

**Review by Thoralf-M — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by qrayven — APPROVED** [review]

**Review by abdulmth — APPROVED** [review]

**Review by kwek20 — APPROVED** [review]

**Pull request merged**